rdflib/namespace.py

```diff
@@ -330,6 +330,21 @@ def compute_qname(self, uri, generate=True):
             namespace, name = split_uri(uri)
             namespace = URIRef(namespace)
             prefix = self.store.prefix(namespace)
+            # check to see if there are longer prefixes defined
+            temp_namespace = namespace.toPython()
+            output_prefix = prefix
+            output_namespace = namespace
+            output_name = name
+            for i in xrange(1, len(name) + 1):
```

Comment:
Ok. This is what I expected we would have to do because this implementation produces worst case behavior all the time. Will take a look and see how it can be improved. Do we have any nice pathological cases that I could use for performance testing?